### PR TITLE
Update the repo to support the current version of ebay-kleinanzeigen

### LINKED ISSUE #5: IndexError: list index out of range
Hi, I can not understand what exactly the error is. Does somebody has any idea?
```
[INFO] Running job "echo (trigger: interval[0:02:00], next run at: 2022-04-20 20:21:34 UTC)" (scheduled at 2022-04-20 20:19:34.563367+00:00)

[INFO] Started echo

[ERROR] Job "echo (trigger: interval[0:02:00], next run at: 2022-04-20 20:21:34 UTC)" raised an exception

Traceback (most recent call last):

  File "/usr/local/lib/python3.8/site-packages/apscheduler/executors/base.py", line 125, in run_job

    retval = job.func(*job.args, **job.kwargs)

  File "/app/main.py", line 121, in echo

    last_items[chat_id] = {'last_item': items[0].url, 'search_url': url}

IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by larsf96**
This pull requests adds some improvements to the current implementation.
Ebay seem to have implemented a bot detection, so a couple of headers were added that solve the problem (fixes #5 )

Also, some of the DOM parsing logic has changed

### CODE REVIEW

**Comment by larsf96** [comment]
It's my first pull request, I've just noticed that there are some "old" commits came along here that I used for testing. They can be ignored, I removed the logging in later commits

@okainov Would be nice if you could review and merge

File: main.py [hunk 1]
```diff
@@ -24,62 +26,78 @@
...
 
 class Item:
-    def __init__(self, title, price, torg, url, date, image):
+    def __init__(self, title, price, torg, url, image):
         self.title = title
         self.price = price
         self.torg = torg
         self.url = 'https://www.ebay-kleinanzeigen.de' + url
-        self.date = date
+        #self.date = date
```
Comment (okainov) [inline main.py]: Why do you remove date?

File: main.py [hunk 2]
```diff
@@ -24,62 +26,78 @@
...
 
     text = qq.text
-    # with open(r'C:\Users\Kleinanzeigen\Desktop\qq.html', 'r') as f:
-    #     text = f.read()
+    
 
-    articles = re.findall('<article(.*?)</article', text, re.S)
 
+    articles = re.findall('<article(.*?)</article', text, re.S)
+    log.info('Articles length %s' % len(articles))
```
Comment (okainov) [inline main.py]: Please use f-string instead.

File: main.py [hunk 3]
```diff
@@ -24,62 +26,78 @@
...
+            price_line = 0
         torg = 'VB' in price_line
         price = None
         if prices := re.findall(r'\d+', price_line, re.S):
             price = int(prices[0])
 
-        date = re.findall('aditem-addon">(.*?)</', item, re.S)[0].strip()
-        if '{' in date or '<' in date:
-            continue
+        # date = re.findall('aditem-addon">(.*?)</', item, re.S)[0].strip()
```
Comment (okainov) [inline main.py]: If the code is not needed - remove it

File: main.py [hunk 4]
```diff
@@ -24,62 +26,78 @@
...
+            name = soup_result[0].text
         else:
             continue
 
-        price_line = re.findall('<strong>(.*?)</strong>', item, re.S)[0]
+        price_line = re.findall('aditem-main--middle--price">(.*?)</p>', item, re.S)
+        if len(price_line) > 0:
+            price_line = price_line[0]
+        else:
+            price_line = 0
```
Comment (okainov) [inline main.py]: I think this will crash as regexp is applied to `price_line` at line 85, it cannot be `int`.

**Review by okainov — COMMENTED** [review]
Thanks for PR! Indeed I don't use it any more, as it worked when I needed it and had no need so far, sorry for that.

Some comments here :)

File: main.py (hunk 1 above)
Comment (larsf96) [reply main.py]: I didn't find a place within the article where the date can be grabbed. The format is (often) something like "Today 12:05". We might need more logic to support that

File: main.py (hunk 2 above)
Comment (larsf96) [reply main.py]: I've updated this in the latest commits

File: main.py (hunk 3 above)
Comment (larsf96) [reply main.py]: I've removed this

File: main.py (hunk 4 above)
Comment (larsf96) [reply main.py]: I've changed it to a string, that should work

**Comment by nilsreichardt** [comment]
Any updates on this?

File: main.py (hunk 1 above)
Comment (okainov) [reply main.py]: Ok, I still think date is needed, we can easily parse heute/gestern, and for other blocks it's regular date, so it should work

**Comment by okainov** [comment]
Ok let me merge that and I'll make few adjustments on top. Sorry for taking so long @nilsreichardt @larsf96 thanks for contribution!

**Pull request merged**